bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlMeterHandler.java

```diff
@@ -0,0 +1,324 @@
+            logger.debug("meter with ID {} not initialized", deviceId);
+            return;
+        }
+
+        MeterType meterType = nhcMeter.getType();
+        if (meterType != MeterType.ENERGY_LIVE) {
+            logger.debug("meter with ID {} does not support live readings", deviceId);
+            return;
+        }
+
+        if (power == null) {
+            updateState(CHANNEL_POWER, UnDefType.UNDEF);
+        } else {
+            boolean invert = getConfig().as(NikoHomeControlMeterConfig.class).invert;
+            int value = (invert ? -1 : 1) * power;
+            updateState(CHANNEL_POWER, new QuantityType<>(value, Units.WATT));
+        }
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void meterReadingEvent(double meterReading, double meterReadingDay, LocalDateTime lastReadingUTC) {
+        NhcMeter nhcMeter = this.nhcMeter;
+        if (nhcMeter == null) {
+            logger.debug("meter with ID {} not initialized", deviceId);
+            return;
+        }
+
+        NikoHomeControlBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null) {
+            logger.debug("Cannot update meter channels, no bridge handler");
+            return;
+        }
+        ZonedDateTime lastReading = lastReadingUTC.atZone(ZoneOffset.UTC)
+                .withZoneSameInstant(bridgeHandler.getTimeZone());
+
+        boolean invert = getConfig().as(NikoHomeControlMeterConfig.class).invert;
+        double value = (invert ? -1 : 1) * meterReading;
+        double dayValue = (invert ? -1 : 1) * meterReadingDay;
+
+        MeterType meterType = nhcMeter.getType();
+        switch (meterType) {
+            case ENERGY_LIVE:
+            case ENERGY:
+                updateState(CHANNEL_ENERGY, new QuantityType<>(value, KILOWATT_HOUR));
+                updateState(CHANNEL_ENERGY_DAY, new QuantityType<>(dayValue, KILOWATT_HOUR));
+                updateState(CHANNEL_ENERGY_LAST, new DateTimeType(lastReading));
+                updateStatus(ThingStatus.ONLINE);
+            case GAS:
+                updateState(CHANNEL_GAS, new QuantityType<>(value, SIUnits.CUBIC_METRE));
+                updateState(CHANNEL_GAS_DAY, new QuantityType<>(dayValue, SIUnits.CUBIC_METRE));
+                updateState(CHANNEL_GAS_LAST, new DateTimeType(lastReading));
+                updateStatus(ThingStatus.ONLINE);
+                break;
+            case WATER:
+                updateState(CHANNEL_WATER, new QuantityType<>(value, SIUnits.CUBIC_METRE));
+                updateState(CHANNEL_WATER_DAY, new QuantityType<>(dayValue, SIUnits.CUBIC_METRE));
+                updateState(CHANNEL_WATER_LAST, new DateTimeType(lastReading));
+                updateStatus(ThingStatus.ONLINE);
+                break;
+            default:
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "@text/offline.configuration-error.meterType");
```

Comment:
Maybe rather something to test in `startCommunication` ?

Reply:
Yes, could indeed be moved over there.